setup.py

```diff
@@ -30,7 +30,7 @@
     python_requires=">=3.7",
     install_requires=[
         "requests>=2.25",
-        "pystac~=1.2.0"
+        "pystac>=1.2.0"
```

Comment:
```suggestion
        "pystac>=1.4.0"
```

We'll need at least 1.4.0 if we want to unblock #142.

Reply:
Hm, good point. @matthewhanson how do you feel about dropping support for pystac < 1.4?

Reply:
Ok to be clear, this PR kept the PySTAC dependency at `pystac >= 1.2.0`. We'll need a separate PR if we want to bump to 1.4.0.